src/main/java/com/redhat/devtools/intellij/knative/actions/func/InvokeAction.java

```diff
@@ -0,0 +1,96 @@
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc.
+ ******************************************************************************/
+package com.redhat.devtools.intellij.knative.actions.func;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.redhat.devtools.intellij.common.utils.ExecHelper;
+import com.redhat.devtools.intellij.knative.actions.KnAction;
+import com.redhat.devtools.intellij.knative.kn.Function;
+import com.redhat.devtools.intellij.knative.kn.Kn;
+import com.redhat.devtools.intellij.knative.telemetry.TelemetryService;
+import com.redhat.devtools.intellij.knative.tree.KnFunctionNode;
+import com.redhat.devtools.intellij.knative.tree.ParentableNode;
+import com.redhat.devtools.intellij.knative.ui.invokeFunc.InvokeDialog;
+import com.redhat.devtools.intellij.knative.utils.model.InvokeModel;
+import com.redhat.devtools.intellij.telemetry.core.service.TelemetryMessageBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.swing.tree.TreePath;
+
+import java.io.IOException;
+
+import static com.redhat.devtools.intellij.knative.telemetry.TelemetryService.NAME_PREFIX_MISC;
+import static com.redhat.devtools.intellij.telemetry.core.util.AnonymizeUtils.anonymizeResource;
+
+public class InvokeAction extends KnAction {
+    private static final Logger logger = LoggerFactory.getLogger(InvokeAction.class);
+    private TelemetryMessageBuilder.ActionMessage telemetry;
+
+    public InvokeAction() {
+        super(KnFunctionNode.class);
+    }
+
+    @Override
+    public void actionPerformed(AnActionEvent anActionEvent, TreePath path, Object selected, Kn knCli) {
+        telemetry = createTelemetry();
+        ParentableNode node = getElement(selected);
+        String name = node.getName();
+        String namespace = knCli.getNamespace();
+        Function function = ((KnFunctionNode) node).getFunction();
+
+        if (function.getLocalPath().isEmpty()) {
+            telemetry
+                    .result(anonymizeResource(name, namespace, "Function " + name + "is not opened locally"))
+                    .send();
+            return;
+        }
+
+        InvokeModel model = new InvokeModel();
+        model.setNamespace(namespace);
+        model.setPath(function.getLocalPath());
+
+        InvokeDialog dialog = new InvokeDialog("Invoke", getEventProject(anActionEvent), function, model);
+        dialog.show();
+
+        if (dialog.isOK()) {
+            ExecHelper.submit(() -> {
+                try {
+                    knCli.invokeFunc(model);
+                    telemetry
+                            .result(anonymizeResource(name, namespace, "Invoked function " + name))
+                            .send();
+                } catch (IOException e) {
+                    logger.warn(e.getLocalizedMessage());
```

Comment:
The exception is object is missing from the method parameters, it is useful as the whole stacktrace will be added to the log if you pass it.